DEVOP-1668 Improve simplified script (#15)
HorizenOfficial/compose-evm-simplified: docs/FORGER.md

> # Forger Node

Reviewer: Comment for file deployments/.gitignore. I think we need to add 
```
*
!.gitignore
```
to be sure that a deployment doesn't cause unstaged changes in git.

## Setup

--- 

Run the init.sh script to initialize the deployment for the first time. Select **forger** node and the **network** to run (eon or gobi).

```shell
./scripts/init.sh
```

The script will generate the required deployment files under the [deployments](../deployments) directory and provide instructions on how to run the compose stack.

## Zend seed

--- 

> As this is a forger node, it requires a zend node to run as well. Syncing a zend node from scratch may take a few hours, 

Reviewer 2: How about this: "Forger node requires zend node to be running as well."

> so a seed file can be used to speed up the process.

Reviewer 2: Replace 'so' with `therefore'

There are two variables that help controlling the seed process for the zend node:

- `ZEN_USE_SEED_FILE`: The default value for this variable is `false`, but it can be set to true choosing to use the seed file in the init.sh script.

- `ZEN_FORCE_RESEED`: This variable is set to `false` by default, but it can be set manually to `true` if forcing the reseed of the zend node is required.

> Once the zend node has been synced, the following times the stack is started there is no need to seed the zend node again.

Reviewer 2: ```suggestion
After the zend node has been synced, there is no need to seed it again when the stack is started subsequently.
```

The _seed.sh_ script will generate a **.seed.complete** file in the node's data folder, and it will prevent the seed script to be run again.

> Additionally, setting the `ZEN_USE_SEED_FILE` variable to false in the .env file will prevent the seed process to be run again.

Reviewer 2: `false` to match your previous style

> If migrating from previous versions of these project and your zend node is already synced set the `ZEN_USE_SEED_FILE` variable to false in the .env file.

Reviewer 2: `false`

> If reseed of the zend node is required, set the `ZEN_FORCE_RESEED` variable to true in the .env file, and restart the stack.

Reviewer 2: `true`

## Running the stack

--- 

1. Prerequisites

>     - Storage: A minimum of 250GB of free space is required to run both nodes. Keep in mind that the storage requirements will grow over time.

Reviewer 3: It could be useful to mention, instead of "both nodes", that we're referring to run "a forger node for both, testnet and mainnet network", just to not get people confused on running "both rpc and forger node".

2. Run the zend node and let it sync (only required the first time the stack is started):
   ```shell
    cd deployments/forger/[eon|gobi] && docker compose up -d zend
    ```

> 3. Verify if the zend node is fully synced you can run the following command and compare the output with the current block height in the mainchain: https://explorer.horizen.io or https://explorer-testnet.horizen.io:

Reviewer 2: ```suggestion
3. Verify if zend node is fully synced by running the following command and comparing the output with the current block height in the mainchain: https://explorer.horizen.io or https://explorer-testnet.horizen.io:
```

    docker exec ${ZEND_CONTAINER_NAME} gosu user curl -s --data-binary '{\"jsonrpc\": \"1.0\", \"id\":\"curltest\", \"method\": \"getblockcount\", \"params\": [] }' -H 'content-type: text/plain;' -u ${ZEN_RPC_USER]}:${ZEN_RPC_PASSWORD} http://127.0.0.1:${ZEN_RPC_PORT}/
    ```

4. Once the zend node is fully synced, run the evmapp node:
    ```shell
    cd deployments/forger/[eon|gobi] && docker compose up -d
    ```

5. Once the evmapp node is fully synced, generate the keys required to run a forger node:
    ```shell
    ./scripts/forger/generate_keys.sh
    ```

   The script will generate all the necessary keys on the local node and output the following:
    ```shell
      Generated VRF Key Pair.
      VRF Public Key         : ...
      VRF Private Key        : ...
    
      Generated Block Sign Key Pair.
      Block Sign Public Key  : ...
      Block Sign Private Key : ...
    
      Generated Ethereum Address Key Pair.
      Ethereum Address                    : ...
      Ethereum Private Key                : ...
      Ethereum Private Key for MetaMask  : ...
    ```
   **STORE THESE VALUES IN A SAFE PLACE. THESE VALUES WILL BE REQUIRED IN THE STAKING PROCESS**

>    Verify that the keys were generated correctly, run the following command:

Reviewer 2: ```suggestion
   Verify that the keys were generated correctly by running the following command:
```
